### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by st-pasha**
- Added documentation for several datatable functions;
- Slightly adjusted style of inline code elements: now missing links are shown in red;
- `:mod:'datatable'` target now works properly.

### CODE REVIEW

File: docs/api/api-index.rst [hunk 1]
```diff
@@ -1,10 +1,78 @@
...
+**Classes**
+--------------------------------------------------------------------------------
+:class:`Frame`             Main data table class
+:class:`ltype`
+:class:`stype`
+
+**Functions**
+--------------------------------------------------------------------------------
+:func:`fread()`            Read CSV/text/XLSX/Jay/other files
+:func:`iread()`            Same, but read multiple files at once
```
Comment (oleksiyskononenko) [inline docs/api/api-index.rst:17]: "Same as `fread()`, ..." would sound better to me.

File: docs/api/api-index.rst [hunk 2]
```diff
@@ -1,10 +1,78 @@
 
+.. py:module:: datatable
+
+datatable API
+=============
+
+=========================  =====================================================
+**Classes**
+--------------------------------------------------------------------------------
+:class:`Frame`             Main data table class
```
Comment (oleksiyskononenko) [inline docs/api/api-index.rst:10]: "datatable" instead of "data table"?

File: docs/api/api-index.rst (hunk 1 above)
Comment (st-pasha) [reply docs/api/api-index.rst:17]: ok

File: docs/api/api-index.rst (hunk 2 above)
Comment (st-pasha) [reply docs/api/api-index.rst]: updated

**Pull request merged**